modules/cms-kit/src/Volo.CmsKit.Admin.HttpApi/Volo/CmsKit/Admin/Blogs/BlogPostAdminController.cs

```diff
@@ -0,0 +1,115 @@
+    {
+        protected readonly IBlogPostAdminAppService BlogPostAdminAppService;
+
+        public BlogPostAdminController(IBlogPostAdminAppService blogPostAdminAppService)
+        {
+            BlogPostAdminAppService = blogPostAdminAppService;
+        }
+
+        [HttpPost]
+        [Authorize(CmsKitAdminPermissions.BlogPosts.Create)]
+        public virtual Task<BlogPostDto> CreateAsync(CreateUpdateBlogPostDto input)
+        {
+            return BlogPostAdminAppService.CreateAsync(input);
+        }
+
+        [HttpDelete]
+        [Route("{id}")]
+        [Authorize(CmsKitAdminPermissions.BlogPosts.Delete)]
+        public virtual Task DeleteAsync(Guid id)
+        {
+            return BlogPostAdminAppService.DeleteAsync(id);
+        }
+
+        [HttpGet]
+        [Route("{id:Guid}")]
+        [Authorize(CmsKitAdminPermissions.BlogPosts.Default)]
+        public virtual Task<BlogPostDto> GetAsync(Guid id)
+        {
+            return BlogPostAdminAppService.GetAsync(id);
+        }
+
+        [HttpGet]
+        [Route("{urlSlug}")]
+        [Authorize(CmsKitAdminPermissions.BlogPosts.Default)]
+        public virtual Task<BlogPostDto> GetByUrlSlugAsync(string urlSlug)
+        {
+            return BlogPostAdminAppService.GetByUrlSlugAsync(urlSlug);
+        }
+
+        [HttpGet]
+        [Route("{id}/cover-image")]
+        [Authorize(CmsKitAdminPermissions.BlogPosts.Default)]
+        public virtual Task<RemoteStreamContent> GetCoverImageAsync(Guid id)
+        {
+            Response.Headers.Add("Content-Disposition", $"inline;filename=\"{id}\"");
+            Response.Headers.Add("Accept-Ranges", "bytes");
+            Response.Headers.Add("Cache-Control", "max-age=120");
+            Response.ContentType = "image/*";
+
+            return BlogPostAdminAppService.GetCoverImageAsync(id);
+        }
+
+        [HttpGet]
+        [Authorize(CmsKitAdminPermissions.BlogPosts.Default)]
+        public virtual Task<PagedResultDto<BlogPostDto>> GetListAsync(PagedAndSortedResultRequestDto input)
+        {
+            return BlogPostAdminAppService.GetListAsync(input);
+        }
+
+        [NonAction]
+        public virtual Task SetCoverImageAsync(Guid id, RemoteStreamContent streamContent)
+        {
+            return BlogPostAdminAppService.SetCoverImageAsync(id, streamContent);
+        }
+
+        [HttpPost]
+        [Route("{id}/cover-image")]
+        [Authorize(CmsKitAdminPermissions.BlogPosts.Update)]
+        public virtual async Task<IActionResult> UploadCoverImageAsync(Guid id, IFormFile file)
+        {
+            if (file == null)
+            {
+                return BadRequest();
+            }
+
+            using (var stream = file.OpenReadStream())
+            {
+                await BlogPostAdminAppService.SetCoverImageAsync(id, new RemoteStreamContent(stream));
+            }
+
+            return StatusCode(201);
```

Comment:
`return Created();`

Reply:
`Created()` methor required 2 parameters at least (`uri`, `object`), So I don't want to return entire byte array back at this situation

https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.controllerbase.created?view=aspnetcore-5.0

Reply:
https://docs.microsoft.com/en-us/aspnet/core/web-api/action-return-types?view=aspnetcore-5.0#asynchronous-action-1 
Can we use `return CreatedAtAction`?